extensions/hdrhistogram/src/main/java/org/radargun/stats/HistogramOperationStats.java

```diff
@@ -250,4 +249,62 @@ private void readObject(ObjectInputStream s) throws IOException, ClassNotFoundEx
       }
       compacted = new Histogram(ranges, counts);
    }
+
+   public final class InternalHistogram {
+
+      private long maxValue;
+      private int digits;
+
+      private org.HdrHistogram.Histogram responseTime;
+      private org.HdrHistogram.Histogram serviceTime;
+      private org.HdrHistogram.Histogram waitTime;
+
+      public InternalHistogram(long maxValue, int digits) {
+         this.maxValue = maxValue;
+         this.digits = digits;
+
+         responseTime = new org.HdrHistogram.Histogram(maxValue, digits);
+         serviceTime = new org.HdrHistogram.Histogram(maxValue, digits);
+         waitTime = new org.HdrHistogram.Histogram(maxValue, digits);
+      }
+
+      public void recordValue(long startTimeNs, long endTimeNs, long intendedTimeNs) {
+         serviceTime.recordValue(endTimeNs - startTimeNs);
+
+         if (intendedTimeNs != 0) { // if intended time is set we can compute response/wait
+            responseTime.recordValue(endTimeNs - intendedTimeNs);
+            waitTime.recordValue(startTimeNs - intendedTimeNs);
```

Comment:
waitTime is never presented, and by definition it's just difference between the other two histograms.